Code/GraphMol/StructChecker/StructChecker.h

```diff
@@ -0,0 +1,266 @@
+//
+//  Copyright (C) 2016 Novartis Institutes for BioMedical Research
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#pragma once
+#ifndef _RD_STRUCTCHECKER_H__
+#define _RD_STRUCTCHECKER_H__
+
+#include <string>
+#include <vector>
+#include "../RDKitBase.h"
+
+/* Example of Usage
+1)  StructChecker chk;
+    int flags = StructureCheck::checkMolStructure( mol ); // use defaults
+ or
+2)
+StructureCheck::StructCheckerOptions options;   // use defaults
+    // To use external data
+    StructureCheck::loadOptionsFromFiles(options, file1, file2, �);
+    StructChecker chk(options);
+
+    for( mol in mols ) {
+        int flags = StructureCheck::checkMolStructure( mol, &options);
+        if (0!=(flags & StructureCheck::StructureFlags::BAD_SET)) {
+        // write to error file
+        } else if (0!=(flags & StructureCheck::StructureFlags::TRANSFORMED_SET))
+{
+        // input molecule was transformed
+        } else { // flag == NO_CHANGE
+        // no change
+        }
+    }
+*/
+
+namespace RDKit {
+namespace StructureCheck {
```

Comment:
structs and enums are required to set input parameters and receive results. They are hidden at least under namespace. Could you please inform what does "details header" means with an examples?